pkg/k8s/waiter.go

```diff
@@ -109,7 +160,7 @@ func (w *Waiter) Wait(timeout time.Duration) bool {
 					if ok {
 						name = fmt.Sprintf("%s/%v.%v", obj["kind"], obj["name"],
 							obj["namespace"])
-						name = watcher.Canonical(name)
+						name = w.canonical(name)
```

Comment:
I feel like that Sprintf needs to be modified to include a GV, and not just `obj["kind"]`.